### ISSUE TITLE
Sonoff S31 - Setoption19 1 causes "RestartReason":"Software Watchdog"

### ISSUE THREAD

**Issue by vel2000** [arendst/Sonoff-Tasmota#5037]
**Describe the bug**
`Setoption19 1` in the console, causes a reboot with: "RestartReason":"Software Watchdog"
As long as the S31 is **configured as a "Sonoff Basic"**, `setoption 19 1` works as expected.
But as soon as the device is **configured as "S31"**, the said reboot occurs.
I've did a `reset 5`, tried a pre-compiled Binary, as well as a self-compiled (6.4.1.10) one, but makes no difference.

_Also, make sure these boxes are checked [x] before submitting your issue - Thank you!_
- [ x] _Searched the problem in issues and in the wiki_
- [ x] _Hardware used_ :  Sonoff S31
- [x ] _Development/Compiler/Upload tools used_ : platformio
- [ x] _If a pre-compiled release or development binary was used, which one?_ : 6.4.1
- [x ] _You have tried latest release or development binaries?_ : release
- [x ] _Provide the output of command_``status 0`` :
```
07:19:59 CMD: status 0
07:19:59 MQT: sonoff/stat/STATUS = {"Status":{"Module":41,"FriendlyName":["Sonoff"],"Topic":"sonoff","ButtonTopic":"0","Power":1,"PowerOnState":3,"LedState":1,"SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0}}
07:19:59 MQT: sonoff/stat/STATUS1 = {"StatusPRM":{"Baudrate":4800,"GroupTopic":"sonoffs","OtaUrl":"http://thehackbox.org/tasmota/release/sonoff.bin","RestartReason":"Software/System restart","Uptime":"0T00:16:23","StartupUTC":"2019-01-27T00:03:36","Sleep":50,"BootCount":12,"SaveCount":37,"SaveAddress":"FA000"}}
07:19:59 MQT: sonoff/stat/STATUS2 = {"StatusFWR":{"Version":"6.4.1.10(sonoff)","BuildDateTime":"2019.01.26 12:00:45","Boot":31,"Core":"2_4_2","SDK":"2.2.1(cfd48f3)"}}
07:19:59 MQT: sonoff/stat/STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["ASUS_chatkaew2",""],"TelePeriod":300,"SetOption":["000A8009","558180C0","00000200"]}}
07:19:59 MQT: sonoff/stat/STATUS4 = {"StatusMEM":{"ProgramSize":458,"Free":544,"Heap":24,"ProgramFlashSize":1024,"FlashSize":4096,"FlashChipId":"1640EF","FlashMode":3,"Features":["00000407","0F402780","24010000","000000C6","000028C0"]}}
07:19:59 MQT: sonoff/stat/STATUS5 = {"StatusNET":{"Hostname":"sonoff-3209","IPAddress":"192.168.2.99","Gateway":"192.168.2.100","Subnetmask":"255.255.255.0","DNSServer":"192.168.2.100","Mac":"5C:CF:7F:7D:2C:89","Webserver":2,"WifiConfig":5}}
07:19:59 MQT: sonoff/stat/STATUS6 = {"StatusMQT":{"MqttHost":"192.168.2.141","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_7D2C89","MqttUser":"DVES_USER","MqttType":1,"MAX_PACKET_SIZE":1000,"KEEPALIVE":15}}
07:19:59 MQT: sonoff/stat/STATUS7 = {"StatusTIM":{"UTC":"Sun Jan 27 00:19:59 2019","Local":"Sun Jan 27 07:19:59 2019","StartDST":"Sun Mar 31 02:00:00 2019","EndDST":"Sun Oct 27 03:00:00 2019","Timezone":"+07:00","Sunrise":"06:42","Sunset":"18:14"}}
07:19:59 MQT: sonoff/stat/STATUS9 = {"StatusPTH":{"PowerDelta":80,"PowerLow":0,"PowerHigh":0,"VoltageLow":0,"VoltageHigh":0,"CurrentLow":0,"CurrentHigh":0}}
07:19:59 MQT: sonoff/stat/STATUS10 = {"StatusSNS":{"Time":"2019-01-27T07:19:59","ENERGY":{"TotalStartTime":"2019-01-27T06:27:39","Total":0.039,"Yesterday":0.000,"Today":0.039,"Power":0,"ApparentPower":0,"ReactivePower":0,"Factor":0.00,"Voltage":224,"Current":0.000}}}
07:19:59 MQT: sonoff/stat/STATUS11 = {"StatusSTS":{"Time":"2019-01-27T07:19:59","Uptime":"0T00:16:23","Vcc":3.497,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"POWER":"ON","Wifi":{"AP":1,"SSId":"ASUS_chatkaew2","BSSId":"D8:50:E6:B3:46:D8","Channel":1,"RSSI":100}}}
```

**To Reproduce**
input `SetOption19 1` into web console

**Expected behavior**
Discoverable switch/sensor data in HA

**Screenshots**
n/a

**Additional context**
**Workaround**: Configure the device temporary as a "Sonoff Basic", use `setoption19 1/0` and than reconfigure it as "S31".
The settings will stick

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by ascillato** [comment]
This was solved in the latest version, please update to latest Tasmota (6.4.1.11) Thanks

**Comment by vel2000** [comment]
Yes, I can confirm, the issue is solved with **6.4.1.11**.
Thanks.

**Comment by ascillato2** [comment]
Great. Thanks for reporting back and testing. Very appreciated. :+1:
